devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -548,12 +544,8 @@ private void triggerCompile(boolean test, boolean prepareNeeded) throws MojoExec
             triggerPrepare(test);
         }
 
-        // compile the Kotlin sources if needed
-        executeIfConfigured(ORG_JETBRAINS_KOTLIN, KOTLIN_MAVEN_PLUGIN, test ? "test-compile" : "compile",
-                Collections.emptyMap());
```

Comment:
Will `process-classes` trigger the compilation in the Kotlin plugin?

Reply:
I don't think this change is correct, see also https://github.com/quarkusio/quarkus/pull/22868/files#r786323138

Reply:
It should yes, but I will add a specific test with a kotlin app.